dask/dataframe/io/parquet/arrow.py

```diff
@@ -274,6 +418,17 @@ def read_metadata(
                             )
                         s["columns"].append(d)
                 s["total_byte_size"] = row_group.total_byte_size
+                if col_chunk_paths:
+                    s["file_path_0"] = row_group.column(0).file_path
+                    if not split_row_groups and (s["file_path_0"] == path_last):
+                        # Rather than appending a new "row-group", just merge
+                        # new `s` statistics into last element of `stats`.
```

Comment:
Is this specific unusual case tested, e.g., having the row-groups in a file out-of-order?

Reply:
Not sure I follow.  This case occurs when the parquet-dataset `pieces` correspond to files, and each file contains multiple row-groups (a scenario that *is* tested now).  In this case, we will have statistics for each row-group, so we need to convert them into file-level statistics for proper filtering etc.

Reply:
Right, but I'm wondering whether the order of the row-groups within a file might be important, if we should also tests a case where they are reversed.

Reply:
Ah - I think I see your point.  If there are multiple row-groups in the same file, the order of those row groups does not matter if we are doing file-level filtering (`split_row_groups=False`), but *does* matter if we want filtering on the row-group level. 
So, in this particular code block, there is no danger of filtering the wrong row-groups.  If we were using `split_row_groups=True`, we could theoretically filter the wrong row-group if the intra-file ordering does not match that of `pieces`.  Hoever, since we are using the builtin `sorted` operation, the sort should be stable.  Therefore, I cannot think of a practical case that the order would differ (I could certainly be wrong here).

Reply:
I think I agree with you. The max of the file for a given column should certainly be the max of the N maxs.